### DISCUSSION TITLE
Vite + Vercel Functions deployment

### DISCUSSION THREAD

**Discussion by chungweileong94** [vercel/community#1242] [Help]
### Summary

I have been using Vite(React) + Vercel Functions deployment for quite some time. However, one thing that I struggling with is that Vercel Functions is really strict about file structure. For instance:
- The `/api` directory needs to need at the root (where the `package.json` is located)
- No custom `tsconfig.*.json` file, which means I have to use `tsconfig.json`

When I use Vite with Vercel Functions, my file structure looks something like:
```
|- /api
    |- /_         <--- To prevent Vercel from making it a function
    |- ...        <--- Vercel Functions
|- /src           <--- Vite src
|- tsconfig.json  <--- tsconfig for both Vite & Vercel function
|- package.json
|- ...
```
This structure works fine until now, however, I don't like the fact that both Vite & Vercel functions share the same `tsconfig.json` & `package.json`, especially the recent ESM support, which means this setup might break sometime in the future. 

I know that I can technically use a custom tsconfig file for Vite instead, but it might also break VSCode Typescript DX for the Vite app codebase, as far as I know, VSCode doesn't allow us to have custom tsconfig per folder. A similar thing can also be applied to the ESLint setup.

I have been thinking if monorepo setup can solve this issue, but I'm not able to find any example for setting up a monorepo with Vercel Functions. So is someone tried to setup a monorepo with Vercel Functions before? Or if you guys have any ideas/ways to separate the configuration for both app & Vercel Functions?

**Comment by twfarland** [comment]
I'm also stuck on this.

I have a yarn monorepo with turborepo. A UI built with vite, and an api folder with my serverless functions, and some common packages that are used across both.

It all deploys green, but when I actually go to invoke the function, my common packages are not found at runtime.

I also can't find any official info on using monorepo packages in serverless functions. It would be nice to be able to control the build for those like we can for UI builds - it's all a bit opaque as it stands. I'd like to be able to use esbuild.
